app/models/course/assessment.rb

```diff
@@ -110,6 +113,27 @@ def set_defaults
     self.autograded ||= false
   end
 
+  def prevent_reminder_duplication
+    # Randomize the milliseconds of the reminders' datetime to prevent duplication
+    self.start_at += Random.rand(0...1.0).round(1) if start_at_changed?
```

Comment:
Shall we make this even smaller, like `Random.rand(0...0.1).round(4)` ?  `round(1)` only returns 10 different values.